### PR TITLE
Div icon

### PR DESCRIPTION
**Pull request by ocefpaf**
@BibMartin this is my first attempt to dive in the new code.  So I have a few questions:

- The original `DivIcon` class was inheriting from `MacroElement`.  I changed it to `Icon`, although I don't think it matters in this case.
- You carefully tried to preserve the method `div_markers`, but I am starting to think that should go away.  It is not really what the original `leaflet` `divIcon` is all about.
- If we agree that `div_markers` should ho we can safely remove the `static_div_icon.js` and the tests for that method.
- I believe there more `<templates>.js` that we can remove.

Note that in this PR I already removed the template.  To see the failure keeping that go to:

https://travis-ci.org/ocefpaf/folium/jobs/91228684

:fr: I am praying for the victims of this horrible attack.  I Hope everybody close to you are alright. :fr:

### CODE REVIEW

**Comment by ocefpaf** [comment]
This PR is the first one of a long term plan to get us closer to matpltolib via mplexporter and mplleaflet.

See: http://nbviewer.ipython.org/gist/ocefpaf/ab173ae7a7ee40430cdf

**Comment by BibMartin** [comment]
> :fr: I am praying for the victims of this horrible attack. I Hope everybody close to you are alright. :fr:

Thank you for your prayers ; everyone I know is fine, but I have a deep thought for past and future victims.

**Comment by BibMartin** [comment]
> This PR is the first one of a long term plan to get us closer to matpltolib via mplexporter and mplleaflet.

That's good idea ; I guess we can end-up with a `MatplotlibFigure` object that does mplleaflet's job.
I wonder whether we shall do this in calling mplleaflet, or get inspiration from it and do it in folium's style.

**Comment by themiurgo** [comment]
Great PR @ocefpaf . I love the notebook with arrow DivIcons, you just gave me a great idea to visualise GPS trajectories. :+1: 

@BibMartin There's nothing else to add to what has been already said in these days. Let's all hope in a better future. :fr:

**Comment by ocefpaf** [comment]
> you just gave me a great idea to visualise GPS trajectories.

Glad that was helpful!  My real intent is to show ocean currents though :wink:  (Soon in a notebook example.)

**Comment by ocefpaf** [comment]
> That's good idea ; I guess we can end-up with a MatplotlibFigure object that does mplleaflet's job.
I wonder whether we shall do this in calling mplleaflet, or get inspiration from it and do it in folium's style.

I like the idea of a `MatplotlibFigure`.  I think that the best course of action is to factor out the `fig2json`.
The fig2json is pretty simple and the only dependency is `mplexporter` and of course `matplolitb`.

PS: I removed the `div_markers` method and tests.  I added some tests for the new `DivIcon`.
Looking at this again I really don't think we should inherent from icon.  Most of we inherent is not applicable to the `DivIcon` (rotation, color, etc).  What do you think?  Should I just inherit from `MacroElement`?

**Comment by ocefpaf** [comment]
> - The original `DivIcon` class was inheriting from `MacroElement`.  I changed it to `Icon`, although I don't think it matters in this case.

@BibMartin I changed it back to `MacroElement` because, even though divIcon is an Icon, it does not make sense to expose the `Icon` methods here.

Let me know how bad this is and if we can salvage some ideas for a better PR :grimacing:

File: folium/features.py [hunk 1]
```diff
@@ -363,39 +364,73 @@ def render(self, **kwargs):
...
+                    {% if this.icon_size %}iconSize: [{{this.icon_size[0]}},{{this.icon_size[1]}}],{% endif %}
+                    {% if this.icon_anchor %}iconAnchor: [{{this.icon_anchor[0]}},{{this.icon_anchor[1]}}],{% endif %}
+                    {% if this.popup_anchor %}popupAnchor: [{{this.popup_anchor[0]}},{{this.popup_anchor[1]}}],{% endif %}
+                    {% if this.className %}className: '{{this.className}}',{% endif %}
+                    {% if this.html %}html: '{{this.html}}',{% endif %}
                     });
                 {{this._parent.get_name()}}.setIcon({{this.get_name()}});
             {% endmacro %}
-            """)
+            """)  # noqa
```
Comment (BibMartin) [inline folium/features.py]: What does `# noqa` mean ?

Comment (ocefpaf) [inline folium/features.py]: No questions asked.  It serves to tell humans and automatic tools to not mess with that line.

**Comment by BibMartin** [comment]
@ocefpaf 
It looks fine to me. I just don't understand why you've dropped out the `class` structure in `test_features.py` and unstacked everything in different functions. But I guess this is not the *active* part of the PR.

> Let me know how bad this is and if we can salvage some ideas for a better PR :grimacing:

It's good to me. Just maybe I should write proper code so that you don't need to pep8 and docstring it.

**Comment by ocefpaf** [comment]
> I just don't understand why you've dropped out the `class` structure in `test_features.py` and unstacked everything in different functions.

Something about `pytest` was not working properly.  I can't remember what.  However, I believe that `pytest` is designed to work better that way.

I will wait for #261 and #262.  (Then I will go back to this one, rebase, polish, and ping you again :smile:)

**Comment by ocefpaf** [comment]
Rebased and good for another round if review.

**Comment by BibMartin** [comment]
Merged : thanks @ocefpaf

**Pull request merged**